samples/hello-cube.html

```diff
@@ -0,0 +1,336 @@
+    const vertexStageDescriptor = {
+        module: shaderModule,
+        entryPoint: "vertex_main"
+    };
+    const fragmentStageDescriptor = {
+        module: shaderModule,
+        entryPoint: "fragment_main"
+    };
+    const pipelineDescriptor = {
+        layout: pipelineLayout,
+
+        vertexStage: vertexStageDescriptor,
+        fragmentStage: fragmentStageDescriptor,
+
+        primitiveTopology: "triangleList",
+        // *GPURasterizationStateDescriptor rasterizationState;
+        // sequence<*GPUBlendStateDescriptor> blendStates;
+        // *GPUDepthStencilStateDescriptor depthStencilState: depthStateDescriptor
+        inputState: inputStateDescriptor
+        // *GPUAttachmentsStateDescriptor attachmentsState;
+ 
+        // u32 sampleCount; // Multisampling
+    };
+    pipeline = device.createRenderPipeline(pipelineDescriptor);
+
+    let colorAttachment = {
+        // attachment is acquired in render loop.
+
+        // GPUTextureView? resolveTarget;
+
+        // *GPULoadOp loadOp;
+        // *GPUStoreOp storeOp;
+        clearColor: { r: 0.5, g: 1.0, b: 1.0, a: 1.0 } // GPUColor
+    };
+
+    // Depth stencil texture
+
+    // *GPUExtent3D
+    // const depthSize = {
+    //     u32 width;
+    //     u32 height;
+    //     u32 depth;
+    // };
+
+    // *GPUTextureDescriptor
+    // const depthTextureDescriptor = {
+    //     *GPUExtent3D size: depthSize,
+    //     u32 arraySize;
+    //     u32 levelCount;
+    //     u32 sampleCount: 1,
+    //     *GPUTextureDimension dimension: "2d",
+    //     *GPUTextureFormat format: "D32FloatS8Uint",
+    //     *GPUTextureUsageFlags usage: GPUTextureUsage.OUTPUT_ATTACHMENT
+    // };
+
+    // GPUTexture + *GPUDevice::createTexture()
+    // const depthTexture = device.createTexture(depthTextureDescriptor);
+
+    // *GPURenderPassDepthStencilAttachmentDescriptor
+    // const depthAttachment = {
+    //     GPUTextureView attachment: depthTexture
+
+    //     *GPULoadOp depthLoadOp;
+    //     *GPUStoreOp depthStoreOp;
+    //     float clearDepth: 1.0;
+
+    //     *GPULoadOp stencilLoadOp;
+    //     *GPUStoreOp stencilStoreOp;
+    //     u32 clearStencil;
+    // };
+
+    renderPassDescriptor = { 
+        colorAttachments: [colorAttachment] 
+        // *depthStencilAttachment: depthAttachment
+    };
+
+    queue = device.getQueue();
+
+    render();
+}
+
+function render() {
+    populateTransformArray(mappedTransformArray);
+
+    const commandBuffer = device.createCommandBuffer();
+    renderPassDescriptor.colorAttachments[0].attachment = context.getNextTexture().createDefaultTextureView();
+    const passEncoder = commandBuffer.beginRenderPass(renderPassDescriptor);
+
+    // Encode drawing commands.
+    passEncoder.setPipeline(pipeline);
+    // Vertex attributes
+    passEncoder.setVertexBuffers(0, [verticesBuffer], [0]);
+    // Bind groups
+    passEncoder.setBindGroup(1, uniformsBindGroup);
```

Comment:
I don't see where you populate bind group 0 with the transform bind group, nor where you create it.

Reply:
Yeah this should be setting bindgroup 0 instead of 1

Reply:
It's setting bindGroup 1 because I haven't added in the logic to allow bind group and vertex buffer indices to overlap on the Metal side. This corresponds to [[buffer(1)]] in the shader since setVertexBuffer(0) is called underneath the hood for the vertex input buffer.